purchase_next_reception_date/models/purchase_order.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    next_reception_date = fields.Date(
+        compute="_compute_next_reception_date", store=True
+    )
+
+    @api.depends(
+        "order_line.date_planned",
+        "order_line.product_qty",
+        "order_line.qty_received",
+        "order_line.qty_invoiced",
```

Comment:
Add `product_id` just in case.

```suggestion
        "order_line.product_id",
        "order_line.product_qty",
        "order_line.qty_received",
        "order_line.qty_invoiced",
        "order_line.date_planned",
```